Sources/QuranAudioKit/Reciter/DownloadedRecitersService.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  DownloadedRecitersService.swift
+//
+//
+//  Created by Zubair Khan on 12/31/22.
+//
+
+import Foundation
+
+public class DownloadedRecitersService {
+
+    public init() {
+    }
+
+    public func downloadedReciters(_ allReciters: [Reciter], _ recentReciters: [Reciter]) -> [Reciter] {
+        let fileManager = FileManager.default
+        let audioFiles = FileManager.documentsURL.appendingPathComponent(Files.audioFilesPathComponent)
+        let downloadedRecitersPaths = try? fileManager.contentsOfDirectory(at: audioFiles, includingPropertiesForKeys: nil)
+        if downloadedRecitersPaths == nil {
+            return []
+        }
+        return filterDownloadedReciters(allReciters, recentReciters, downloadedRecitersPaths!)
```

Comment:
Let's avoid `!` if possible. In this case you could just do if let in the previous line.